final/background/spoofax.tex

```diff
@@ -87,211 +119,139 @@ \subsubsection{SDF3: syntax definition in Spoofax}
-transformation with rewrite rules~\cite{Visser01}. Stratego can be
-seen as the most general part of Spoofax: before NaBL and TS, Stratego
-was used for specifying the static semantics. Moreover, being a
-program transformation language, it can also serve as a compiler and
-can thus be used to specify the dynamic semantics.
-
-An example of a rewrite rule for the paplj language is given below.
-\lstset{language=stratego,numbers=left}
-\begin{lstlisting}
-rules
-  desugar-let :
-  	Let([], e) -> e
-
-  desugar-let :
-  	Let([b1, b2 | bs], e) -> Let([b1], Let([b2 | bs], e))
-\end{lstlisting}
-This desugars a \texttt{let} expression with multiple bindings into multiple
-nested \texttt{let} expressions each having just one binding. Again it can be
-seen that these are syntax-directed rules, from the way the rules are
-specified using the AST.
-
-To construct the main algorithm of the program transformation,
-Stratego has the notion of \emph{strategies}. A strategy is used to specify
-where and in what order the rewrite rules are applied to an
-AST. Another example from paplj is given below:
-\lstset{language=stratego,numbers=left}
-\begin{lstlisting}
-strategies
-  pre-desugar =
-    innermost(desugar-let <+ desugar-do)
-
-  post-desugar =
-    innermost(desugar-do <+ desugar-get <+ desugar-set);
-    resugar
-\end{lstlisting}
-The strategy \texttt{innermost} in this example is used to apply the strategy
-given as parameter (a composition of rewrite rules) in a specific
-traversal order on the AST nodes.
-
-The Spoofax API provides the \texttt{TranformService} for performing program
-transformation. Internally the \texttt{TransformService} accesses the
-Stratego runtime, which is retrieved from the
-\texttt{StrategoRuntimeService}. The same holds for the \texttt{AnalysisService} of
-the previous section: it too uses the Stratego runtime.
-
-Stratego furthermore has support for \emph{native} strategies, which are
-specified in Java instead. Therefore the interface is bidirectional:
-Stratego can hook into Java, and Java can use the Stratego API.
+Stratego terms can however represent any tree structure, not just ASTs. As such,
+when using Spoofax programmatically through its API, one sees Stratego terms
+used as an internal representation of the data going through the execution
+pipeline depicted in \cref{fig:spoofax-relations}.
 
 \subsection{Dynamic Semantics}
 \label{ssec:dynamic-semantics}
 Dynamic semantics refers to how a program written in some language
-behaves~\cite{Winskel93}. There are many approaches to formally
-specify the dynamic semantics of a programming language (for an
-extensive treatment, see~\cite{Winskel93}). For this section only
-one sort of approach called \emph{operational semantics} is relevant.
+behaves~\cite{Winskel93}. There are many approaches to formally specify the
+dynamic semantics of a programming language (for an extensive treatment,
+see~\cite{Winskel93}). One way, for example, is to specify the transformation of
```

Comment:
This renders as "see[xx]", which looks weird. Can you maybe say something like "see this paper~\cite{Winskel93}"?